### ISSUE TITLE
Notice: Uninitialized string offset: 0 in ~\humbug\src\Adapter\Locator.php on line 27

### ISSUE THREAD

**Issue by johanneskonst** [humbug/humbug#169]
Ran humbug (added as composer require --dev humbug/humbug) without having tests and got a lot of Notices. Added `!empty($name) &&` to start of if statement in ~\humbug\src\Adapter\Locator.php line 27 and some `(` and `)` to keep the if statement in working order and all is peachy again.

I'm unable to provide a PR for this, so this is just a heads up.

**Comment by theofidry** [comment]
Hi, apologize for the late answer. Could you provide what kind of notices you are getting? I'm not really getting any weird error or at least they seem ok and your changes don't change anything for me.

**Comment by padraic** [comment]
Looks like there's a PR attached. It looks like there would be an issue if a variable was empty, but we still tried searching it as an array.

**Comment by theofidry** [comment]
Sorry, I meant I did try the changes from in #189, but I didn't see any change.

**Comment by padraic** [comment]
I think this basically resolves down to...what if there are zero tests anywhere. Admittedly, never attempted here! Expanding on that, what if phpunit isn't installed? We probably should have reasonable failure messages for those eventualities, no matter how ridiculous they sound.

**Comment by theofidry** [comment]
@padraic I completely agree with that :) It's just that I couldn't reproduce the issue neither see any change after applying the patch.

**Comment by padraic** [comment]
@theofidry If we can't reproduce, we should probably close this and let the problem (if there is one) bubble up another time when there might be a reproducible example to work with. It does seem that if the variable is blank, we're missing something earlier in the execution. This might conceal the error, and then conceal the problem, and make things more confusing down the line.

**Comment by theofidry** [comment]
Let's close it then and make sure to test those kind of weird scenarios to provide meaningful error messages.

**Comment by johanneskonst** [comment]
@theofidry about the late awnser: no worries ;-)

So, FYI:

The notices are indeed the following:
`PHP Notice:  Uninitialized string offset: 0 in ~\vendor\humbug\humbug\src\Adapter\Locator.php on line 27`
`PHP Notice:  Uninitialized string offset: 0 in ~\vendor\humbug\humbug\src\Adapter\Locator.php on line 28`
_(and yes, i run this on windows, php7 with xdebug and display_errors set to showevery*&^%$thing)_
And that repeated a multitude of times. And like @padraic said it is (probably) caused by not having tests. When the PR is added it doesn't fill your screen full with notices but only shows the normal output (of 0 mutations generated) which is what i expect.

Basically, i start my project, setup directories and such, add phpunit, it's xml.dist, test if it works (No tests executed!), add humbug (usually dev-master), it's json.dist, test if it works (0 mutations) and be done/ready (and _not_ be greeted with a few hundred lines of notices, hence my report/pr).

Hope this helps to clarify it a bit. If there are any questions feel free to ask.


Regards,

**Comment by padraic** [comment]
Added a separate issue to cover such a scenario of a project starting from scratch.

**Comment by theofidry** [comment]
Might be a windows thingy as well then, as I did try that scenario :)

Could you try to see if #232 fixes the issue?

**Comment by padraic** [comment]
@johanneskonst With #232 now integrated, and #248 checking scenarios with no tests, and no source code, there are no errors on any platform via the CIs. This all includes a few minor Windows fixes which may have done the trick.

By all means do re-open if the issue has persisted!
